### DISCUSSION TITLE
Incorrect Error Message: `identifier "concept" is undefined`

### DISCUSSION THREAD

**Discussion by lemorage** [microsoft/vscode-cpptools#12156] [Q&A] [answered]
If I put the following code into the vscode editor, it will notify me `identifier "concept" is undefined C/C++(20)` (while I can compile and run it smoothly). I'd like to know if there is a way to get around this. I think some existing issues are pointing to this, like #11624, but I am unsure if this is still to be resolved.
```cpp
#include <concepts>

template<typename Num>
concept Numeric = std::is_arithmetic_v<Num>;

template<typename Num>
requires Numeric<Num>
double convert_f_to_c(Num temperature) {
    return (temperature - 32) * 5 / 9;
}
```

<img width="1137" alt="Screenshot 2024-03-29 at 22 38 23" src="https://github.com/microsoft/vscode-cpptools/assets/88943827/a71613c0-61c5-479a-bee6-065fde48c61c">
<br/><br/>

- MacOS Version: 14.1.2
- VSCode Version: 1.87.2
- C/C++ Extension Version: 1.19.9
- Clang Version: 15.0.0

**Comment by sean-mcmanus** [comment, marked as answer]
@lemorage It works for me without errors. You need to set cppStandard to "c++20" or higher. Also, running the C/C++: Log Diagnostics command with test.cpp active should give more configuration info.

**Reply by lemorage** [reply]
Ah, yes! That works. I thought I'd set the config before, but it turned out the config was the default in this project. That's why the error occurs. Anyway, thank you!
